vike/node/plugin/plugins/importUserCode/getVirtualFileImportUserCode.ts

```diff
@@ -317,10 +323,11 @@ function getGlobs(
 }
 
 function getGlobRoots(config: ResolvedConfig, configVike: ConfigVikeResolved): GlobRoot[] {
+  const gitignoreLines = getGitignoreLines()
   const globRoots: GlobRoot[] = [
     {
       includeDir: '/',
-      excludeDir: path.posix.relative(config.root, getOutDirs(config).outDirRoot)
+      excludeDir: [path.posix.relative(config.root, getOutDirs(config).outDirRoot), ...gitignoreLines]
```

Comment:
This won't work unfortunately because `gitignoreLines` can contain negative patterns like `!something`. But then on line 308 we prefix `excludeDir` also with an exclamation mark `!`. So we will end up with double exclamation marks `!!` being fed to `import.meta.glob()`, which I don't think works.

> Note: At other places however where we just pass all the lines of gitignore to `fast-glob`, things are fine as `fast-glob` [supports negative patterns](https://github.com/mrmlnc/fast-glob)

> Other note: if `excludeDir` can now contains several dirs, it should be renamed `excludeDirs`

Reply:
What if readAndSetGitignoreLines filtered the lines starting with `!`? I think we can ignore those lines.

Reply:
It might go wrong in some cases though:

```
/ignore/this/folder/
!/ignore/this/folder/but/not/this/subfolder/  # we would miss what is in there if we ignore this line
```

But fine with me as this is, I think, quite rare. We could also print a warning in the console if we encounter such lines in the gitignore.

Another possibility is to not care too much, as this is the logic for the old V0.4 design (the `.page` files), not for the V1 design (the `+` files), so maybe it's fine if we don't even support gitignore at all for the old design? What do you think @brillout ?

Reply:
So my vote would be: we don't touch this V0.4 old logic at all. One way or another we can't leave it as is, as if we end up writing `!!` things will fail in the user's project in cryptic ways

Reply:
The underlying issue is https://github.com/vikejs/vike/issues/818 which also holds for the V1 design. If we manage to apply `.gitignore` to Vike's `fast-glob` usage that'd be quite neat as it'd solve the issue in the (vast?) majority of cases.

I wonder if it's feasible to apply `.gitignore` in a safe way, albeit only a subset of `.gitignore`.

Reply:
so would you be fine solving it for V1-only?

Reply:
Yes adding this only for the V1 design is fine.

Reply:
Vite's `import.meta.glob()` uses `fast-glob` so I believe also supporting the V0.4 design doesn't add much complexity. That said, if omitting the V0.4 design makes our life easier, then that'd be more than fine because we're going to release the V1 design in the next coming days and eventualy deprecate the V0.4 design.